### DISCUSSION TITLE
Left Right Channel swapped

### DISCUSSION THREAD

**Discussion by ilinsdell** [schreibfaul1/ESP32-audioI2S#263] [General]
I recently connected my ESP32 to an Adafruit I2S Stereo Decoder - UDA1334A Breakout.

It seems like the code for the channels is swapped in this library?

ESP sets format as: m_i2s_config.channel_format       = I2S_CHANNEL_FMT_RIGHT_LEFT;

yet the playChunk(.) code sends in reverse order:

sample[LEFTCHANNEL]  = x;
sample[RIGHTCHANNEL] = y;

Sure enough when testing, my left channel is comming from the right speaker.

**Comment by schreibfaul1** [comment]
I downloaded a test file from the Fraunhofer Institute https://www2.iis.fraunhofer.de/AAC/SBRtestStereo-441-16b.wav and tested it with a PCM5102a. In fact, left and right are reversed. I didn't notice that with the AC101 DAC. I have corrected the defect. Thanks for the hint.

**Reply by ilinsdell** [reply]
Thanks for response/confirmation.

I applied update and verified channels are working correctly now.

Thanks again
